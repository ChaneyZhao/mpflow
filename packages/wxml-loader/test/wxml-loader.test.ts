/* eslint-env jest */
import compile from './helpers/compile'

describe('wxml-loader', () => {
  test('should work', async () => {
    const result = await compile('basic', {})
    expect(result.exports).toMatchSnapshot('exports')
    expect(result.warnings).toMatchSnapshot('warnings')
    expect(result.errors).toMatchSnapshot('errors')
  })

  test('work with esModule', async () => {
    const result = await compile('basic', { esModule: true })
    expect(result.exports).toMatchSnapshot('exports')
    expect(result.warnings).toMatchSnapshot('warnings')
    expect(result.errors).toMatchSnapshot('errors')
  })

  test('work with minimize', async () => {
    const result = await compile('basic', { minimize: true })
    expect(result.exports).toMatchSnapshot('exports')
    expect(result.warnings).toMatchSnapshot('warnings')
    expect(result.errors).toMatchSnapshot('errors')
  })

  test('work with mustache disabled', async () => {
    const result = await compile('mustache', { minimize: true })
    expect(result.exports).toMatchSnapshot('exports')
    expect(result.warnings).toMatchSnapshot('warnings')
    expect(result.errors).toMatchSnapshot('errors')
  })

  test('work with mustache enabled', async () => {
    const result = await compile('mustache', { minimize: true, resolveMustache: true })
    expect(result.exports).toMatchSnapshot('exports')
    expect(result.warnings).toMatchSnapshot('warnings')
    expect(result.errors).toMatchSnapshot('errors')
  })
})
