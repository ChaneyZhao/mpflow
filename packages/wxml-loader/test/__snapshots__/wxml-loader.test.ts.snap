// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`wxml-loader should work: errors 1`] = `Array []`;

exports[`wxml-loader should work: exports 1`] = `
Object {
  "default": WxmlModule {
    "exports": Array [
      Array [
        3,
        "<template name=\\"item\\">
  <text>{{text}}</text>
</template>
",
        "b.wxml",
        undefined,
      ],
      Array [
        1,
        "<import src=\\"b.wxml\\"/>

<wxs module=\\"foo\\">
var some_msg = \\"hello world\\";
module.exports = {
  msg : some_msg,
}
</wxs>
<wxs module=\\"c\\" src=\\"c.adca9fdd.wxs\\"/>
<view>
  <input type=\\"password\\"/>
  <!-- <form>
    <label>
  </form> -->
  <image src=\\"image.09a1a111.png\\"/>
  {{foo.msg}}
</view>",
        "a.wxml",
        undefined,
      ],
    ],
    "imported": Set {
      3,
      1,
    },
    "url": "a.wxml",
  },
}
`;

exports[`wxml-loader should work: warnings 1`] = `Array []`;

exports[`wxml-loader work with esModule: errors 1`] = `Array []`;

exports[`wxml-loader work with esModule: exports 1`] = `
Object {
  "default": WxmlModule {
    "exports": Array [
      Array [
        3,
        "<template name=\\"item\\">
  <text>{{text}}</text>
</template>
",
        "b.wxml",
        undefined,
      ],
      Array [
        1,
        "<import src=\\"b.wxml\\"/>

<wxs module=\\"foo\\">
var some_msg = \\"hello world\\";
module.exports = {
  msg : some_msg,
}
</wxs>
<wxs module=\\"c\\" src=\\"c.adca9fdd.wxs\\"/>
<view>
  <input type=\\"password\\"/>
  <!-- <form>
    <label>
  </form> -->
  <image src=\\"image.09a1a111.png\\"/>
  {{foo.msg}}
</view>",
        "a.wxml",
        undefined,
      ],
    ],
    "imported": Set {
      3,
      1,
    },
    "url": "a.wxml",
  },
}
`;

exports[`wxml-loader work with esModule: warnings 1`] = `Array []`;

exports[`wxml-loader work with minimize: errors 1`] = `Array []`;

exports[`wxml-loader work with minimize: exports 1`] = `
Object {
  "default": WxmlModule {
    "exports": Array [
      Array [
        3,
        "<template name=\\"item\\"><text>{{text}}</text></template>",
        "b.wxml",
        undefined,
      ],
      Array [
        1,
        "<import src=\\"b.wxml\\"/><wxs module=\\"foo\\">
var some_msg = \\"hello world\\";
module.exports = {
  msg : some_msg,
}
</wxs><wxs module=\\"c\\" src=\\"c.adca9fdd.wxs\\"/><view><input type=\\"password\\"/><image src=\\"image.09a1a111.png\\"/>{{foo.msg}}</view>",
        "a.wxml",
        undefined,
      ],
    ],
    "imported": Set {
      3,
      1,
    },
    "url": "a.wxml",
  },
}
`;

exports[`wxml-loader work with minimize: warnings 1`] = `Array []`;

exports[`wxml-loader work with mustache: errors 1`] = `Array []`;

exports[`wxml-loader work with mustache: exports 1`] = `
Object {
  "default": WxmlModule {
    "exports": Array [
      Array [
        1,
        "<wxs module=\\"$w0\\">module.exports={\\"./img/m-a.png\\":\\"m-a.31d6cfe0.png\\",\\"./img/m-b.png\\":\\"m-b.31d6cfe0.png\\"}</wxs><image src=\\"$w0[./img/m-{{id}}.png]\\"/>",
        "a.wxml",
        undefined,
      ],
    ],
    "imported": Set {
      1,
    },
    "url": "a.wxml",
  },
}
`;

exports[`wxml-loader work with mustache: warnings 1`] = `Array []`;
